react-native/index.d.ts

```diff
@@ -4554,12 +4554,22 @@ declare module "react" {
     // @see https://github.com/facebook/react-native/blob/0.34-stable\Libraries\StyleSheet\StyleSheetTypes.js
     export namespace StyleSheet {
 
+        interface Styles {
+            [style: string]: Style
+        }
+
         type Style = ViewStyle | TextStyle | ImageStyle
 
         /**
          * Creates a StyleSheet style reference from the given object.
          */
-        export function create<T>( styles: T ): T;
+        // Non-generic override is required to provide intellisense
+        // for JavaScript and non-generic method invocations
+        export function create(styles: Styles): any;
```

Comment:
@mhegazy, it sounds like it is, but in fact, making it return typed value breaks accessing stylesheet styles, see example below

![screen shot 2017-01-19 at 11 02 49](https://cloud.githubusercontent.com/assets/3857604/22098496/a03a4dac-de37-11e6-828e-29e507490910.png)